src/common/pf_scheduler.c

```diff
@@ -322,36 +377,37 @@ int pf_scheduler_add (
    }
    os_mutex_unlock (net->scheduler_timeout_mutex);
 
-   *p_timeout = ix_free + 1; /* Make sure 0 is invalid. */
+   handle->timer = ix_free + 1; /* Make sure 0 is invalid. */
 
    return 0;
 }
 
-void pf_scheduler_remove (pnet_t * net, const char * p_name, uint32_t timeout)
+void pf_scheduler_remove (pnet_t * net, pf_scheduler_handle_t * handle)
 {
    uint16_t ix;
 
-   if (timeout == 0)
+   if (handle->timer == 0 || handle->timer > PF_MAX_TIMEOUTS)
    {
-      LOG_DEBUG (
+      LOG_ERROR (
          PNET_LOG,
-         "SCHEDULER(%d): timeout == 0 for event %s. No removal.\n",
+         "SCHEDULER(%d): Invalid value %u for timeout \"%s\". No removal.\n",
          __LINE__,
-         p_name);
+         handle->timer,
+         handle->p_name);
    }
    else
    {
-      ix = timeout - 1; /* Refer to _add() on how p_timeout is created */
+      ix = handle->timer - 1; /* Refer to _add() for handle->timer creation */
       os_mutex_lock (net->scheduler_timeout_mutex);
 
-      if (net->scheduler_timeouts[ix].p_name != p_name)
+      if (net->scheduler_timeouts[ix].p_name != handle->p_name)
```

Comment:
Strange that net->scheduler_timeouts[ix].in_use is not checked but the name which is managed outside the scheduler is checked

Reply:
Added check of `.in_use`